oauth2/src/main/java/com/okta/spring/boot/oauth/env/OktaOAuth2PropertiesMappingEnvironmentPostProcessor.java

```diff
@@ -106,36 +110,63 @@ final class OktaOAuth2PropertiesMappingEnvironmentPostProcessor implements Envir
     @Override
     public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
 
+        OIDCMetadata oidcMetadata;
+        try {
+            String issuer = environment.getProperty(OKTA_OAUTH_ISSUER);
+            if (issuer != null) {
+                RestTemplate restTemplate = new RestTemplate();
+                if (!issuer.endsWith("/")) {
+                    issuer += "/";
+                }
+                ResponseEntity<String> response
+                    = restTemplate.getForEntity(issuer + ".well-known/openid-configuration", String.class);
```

Comment:
Does this lookup happen on startup? When I was working on my native Java examples post, I discovered that Spring Security did a metadata lookup on startup. I convinced them they should do it on the first request instead, so the startup time would be faster. It'd be unfortunate if this reverted the default Spring Security behavior. 

The best way to test this is likely to create two different Spring Boot apps. One uses the Okta Spring Boot starter and one uses raw Spring Security. Compile them both to native images. Then compare startup times.